File: samples/vulnerable/weak_crypto.cs
// VibeGuard sample: C# weak crypto patterns.
using System;
using System.Security.Cryptography;
using System.Text;

namespace VibeGuardSamples
{
    public static class WeakCrypto
    {
        // VG-CRYPTO-001 — MD5 hash of a password.
        public static string LegacyHash(string password)
        {
            using var hasher = MD5.Create();
            var bytes = hasher.ComputeHash(Encoding.UTF8.GetBytes(password));
            return BitConverter.ToString(bytes);
        }

        // VG-CRYPTO-001 — SHA1 used for "integrity".
        public static byte[] LegacyChecksum(byte[] data)
        {
            using var hasher = SHA1.Create();
            return hasher.ComputeHash(data);
        }

        // VG-CRYPTO-002 — System.Random used to generate a token.
        public static int NewToken()
        {
            var token = new Random().Next();
            return token;
        }
    }
}

File: samples/vulnerable/weak_crypto.rb
# VibeGuard sample: Ruby weak crypto patterns.

require 'digest'

# VG-CRYPTO-001 — MD5 fingerprint over user data.
def legacy_fingerprint(payload)
  Digest::MD5.hexdigest(payload)
end

# VG-CRYPTO-001 — SHA1 used as a "secure" hash.
def legacy_signature(data)
  Digest::SHA1.hexdigest(data)
end

# VG-CRYPTO-002 — Kernel#rand for a session id.
def new_session_id
  session_id = rand(1_000_000)
  session_id
end

# VG-CRYPTO-002 — Random.new for a token.
def new_token
  token = Random.new.bytes(32)
  token
end
